File: 0003-MED-longest-substring-without-repeating-chars/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        char[] chars = s.toCharArray();
        StringBuilder substring = new StringBuilder();
        int maxLength = 0;

        for (int i = 0; i < chars.length; i++) {
            if (substring.indexOf(String.valueOf(chars[i])) != -1) {
                int repeatedIndex = substring.indexOf(String.valueOf(chars[i]));
                substring = new StringBuilder(substring.substring(repeatedIndex + 1));
            }
            
            substring.append(chars[i]);
            if (substring.length() > maxLength) {
                maxLength = substring.length();
            }
        }

        return maxLength;
    }
}

File: 0003-MED-longest-substring-without-repeating-chars/solution.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        substring = ""
        max_length = 0

        for i in range(len(s)):
            if s[i] in substring:
                repeated_index = substring.index(s[i])
                substring = substring[repeated_index + 1:]

            substring += s[i]
            max_length = max(max_length, len(substring))

        return max_length